x-pack/plugins/data_enhanced/server/search/es_search_strategy.ts

```diff
@@ -5,114 +5,83 @@
-            ? await asyncSearch(request, options, context)
-            : await rollupSearch(request, options, context);
-
-          if (
-            usage &&
-            isAsync &&
-            isEnhancedEsSearchResponse(response) &&
-            isCompleteResponse(response)
-          ) {
-            usage.trackSuccess(response.rawResponse.took);
-          }
-
-          resolve(response);
-        } catch (e) {
-          if (usage) usage.trackError();
-          reject(e);
-        }
-      })
-    );
-
-  const cancel = async (context: RequestHandlerContext, id: string) => {
-    logger.debug(`cancel ${id}`);
-    await context.core.elasticsearch.client.asCurrentUser.asyncSearch.delete({
-      id,
-    });
-  };
-
-  async function asyncSearch(
+  function asyncSearch(
     request: IEnhancedEsSearchRequest,
     options: ISearchOptions,
     context: RequestHandlerContext
-  ): Promise<IEsSearchResponse> {
-    let promise: TransportRequestPromise<any>;
-    const esClient = context.core.elasticsearch.client.asCurrentUser;
-    const uiSettingsClient = await context.core.uiSettings.client;
+  ) {
     const asyncOptions = getAsyncOptions();
 
-    // If we have an ID, then just poll for that ID, otherwise send the entire request body
-    if (!request.id) {
-      const submitOptions = toSnakeCase({
-        batchedReduceSize: 64, // Only report partial results every 64 shards; this should be reduced when we actually display partial results
-        ...(await getDefaultSearchParams(uiSettingsClient)),
-        ...asyncOptions,
-        ...request.params,
-      });
-
-      promise = esClient.asyncSearch.submit(submitOptions);
-    } else {
-      promise = esClient.asyncSearch.get({
-        id: request.id,
-        ...toSnakeCase(asyncOptions),
-      });
-    }
-
-    const esResponse = await shimAbortSignal(promise, options?.abortSignal);
-    const { id, response, is_partial: isPartial, is_running: isRunning } = esResponse.body;
-    return {
-      id,
-      isPartial,
-      isRunning,
-      rawResponse: shimHitsTotal(response),
-      ...getTotalLoaded(response._shards),
-    };
+    return config$.pipe(
+      mergeMap(
+        () =>
+          new Promise<DoSearchFnArgs>(async (resolve) => {
+            const defaultParams = await getDefaultSearchParams(context.core.uiSettings.client);
+
+            resolve({
+              params: {
+                ...defaultParams,
+                batchedReduceSize: 64,
+                ...asyncOptions,
+                ...request.params,
+              },
+            });
+          })
+      ),
+      switchMap(
+        doPartialSearch(
+          (...args) => context.core.elasticsearch.client.asCurrentUser.asyncSearch.submit(...args),
+          (...args) => context.core.elasticsearch.client.asCurrentUser.asyncSearch.get(...args),
+          request.id,
+          asyncOptions,
+          options
+        )
+      ),
+      esSearch.toKibanaSearchResponse(),
```

Comment:
Can we export all of the methods on `esSearch` in a `utils` directory or something so we don't need to access them from `esSearch`?